### PR TITLE
Test controller

### PR DESCRIPTION
**Pull request by Haywayaheadshot**
# In this milestone I;
> I wrote tests for user_controller and test_controller
> I updated README.md to including testing step.

### CODE REVIEW

File: spec/requests/post_spec.rb [hunk 1]
```diff
@@ -2,6 +2,28 @@
...
+        expect(response).to render_template('index')
+      end
+      it 'should render correct placeholder in index posts page' do
+        get @route
+        expect(response.body.to_json).to include('Post Index HTML File')
+      end
+      it 'should render correct placeholder in the page using assert_select' do
+        get @route
+        assert_select 'h1', 'Post Index HTML File'
+      end
```
Comment (AnselemOdims) [inline spec/requests/post_spec.rb]: - [x] Good job working on this, but please kindly include tests for post show page

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Haywayaheadshot , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Haywayaheadshot** [comment]
@AnselemOdims Thank you for reviewing my project. I will work on the change asap

File: spec/requests/post_spec.rb (hunk 1 above)
Comment (rloterh) [reply spec/requests/post_spec.rb]: resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Abubakar,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Controller specs properly implemented!

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Haywayaheadshot** [comment]
@rloterh Thank you for reviewing my project. Have a nice day.

**Pull request merged**